### DISCUSSION TITLE
Interactive notebook examples

### DISCUSSION THREAD

**Discussion by bobturneruk** [pykale/pykale#147] [Ideas] [answered]
Hi everyone,

I've been looking at how we might make interactive notebook examples (e.g. with Google Collab or myBinder) for some of the examples. It seems like training times of a couple of hours might be fairly typical (perhaps this can be sped up with better tensorflow config?). Given this, I think having notebooks for people to run online is probably not something anyone will have the patience for. I suggest that our use of notebooks might instead be around having things that people can download and run, or look at "statically" as part of online documentation.

Any thoughts on this?

All the best,

Bob

**Comment by haipinglu** [comment, marked as answer]
Thanks Bob. This just confirms the need of tutorials and examples are not serving the purpose well.

We can definitely make the training time bearable (in minutes rather than hours). The solution is to configure the examples to the _fast_ and _compact_ mode. The issue is also similar to testing.

You may start with the cifar or digits example and I can provide input on simplifying the config. We can talk about this on Thursday.

**Reply by bobturneruk** [reply]
OK - I'll have a look at digits - I'd started with deepdta.

**Reply by haipinglu** [reply]
Thanks I'll talk to @pz-white later about deepdta.

**Reply by haipinglu** [reply]
Create a new yaml under `configs` to customize the configuration to a _fast_ one

For deepdta 
- use the smallest `Kd` dataset `Kd-DeepDTA.yaml` will be the fastest. 
- set SOLVER.MAX_EPOCHS to a small number, e.g. 1, 2, 5, or 10 rather than the default 100

For digits
- change DATASET.SOURCE from "mnist" to "svhn"
- change SOLVER.MAX_EPOCHS from 120 to a small number
- change SOLVER.MIN_EPOCHS from 20 to a small number, e.g. 1

**Reply by bobturneruk** [reply]
Thanks!

**Reply by bobturneruk** [reply]
Is there some documentation on how to populate the `.yaml` config files e.g. what headings, names, types are recognised? I can infer a bit from the examples... Sorry if I'm missing something obvious!

**Reply by bobturneruk** [reply]
Also, I've just encountered this - I'll try and fix https://github.com/PyTorchLightning/pytorch-lightning/issues/1314

**Reply by bobturneruk** [reply]
I'd like to try and run without using a GPU device, if possible.

**Reply by bobturneruk** [reply]
Removing this line fixes https://github.com/pykale/pykale/blob/9b9ffcf697f59429ef6a4d957166e65097c1beba/examples/digits_dann_lightn/main.py#L69

**Reply by haipinglu** [reply]
> Is there some documentation on how to populate the `.yaml` config files e.g. what headings, names, types are recognised? I can infer a bit from the examples... Sorry if I'm missing something obvious!

See links at the 6th bullets of https://github.com/pykale/pykale/blob/main/.github/CONTRIBUTING.md#coding-style

**Reply by haipinglu** [reply]
> Removing this line fixes

Yes, tutorials should not require GPU to run. Thanks.

**Reply by bobturneruk** [reply]
Cheers @haipinglu - I feel like more docs on use of `.yaml` may be helpful to users. Something to discuss more, perhaps.

**Reply by haipinglu** [reply]
> Cheers @haipinglu - I feel like more docs on use of `.yaml` may be helpful to users. Something to discuss more, perhaps.

Yes, thanks. A card for it: https://github.com/pykale/pykale/projects/9#card-61402428

**Comment by haipinglu** [comment]
Very good examples here: https://github.com/pytorchbearer/torchbearer
